### PR TITLE
[#10645] Admin search results: show more info

### LINKED ISSUE #10645: Admin search results: show more info
Steps: login as admin and search for a student.

Current:
image

Suggestions:
* Link student name to the course  of that student in the course instructor (i.e., masquerade as the instructor of the course) 
  * Reason: This link was used to quickly reach the session/course details when students send us support requests
* ~Give the full links i.e., starting with `https://teammates...`~
  * ~Reason: I copy-paste these links to preview the submission page, when students request us support requests~
  * Done

### PR DESCRIPTION
**Pull request by g-lilian**
Fixes #10645

Finished the first part of the suggestion in the issue by linking student name to course details page in the admin search results.

*Ready for review

Screenshot_1

### CODE REVIEW

**Comment by g-lilian** [comment]
Strange, the top 3 commits were from my previous PR (already merged). I am not sure why it is appearing in the commit history.

**Comment by rrtheonlyone** [comment]
@g-lilian you can follow the below steps to remove those commits:

```
git fetch upstream
git pull --rebase upstream master
git log --online #check commits are gone 
git push --force origin 10645
```

**Comment by g-lilian** [comment]
> @g-lilian you can follow the below steps to remove those commits:
> 
> ```
> git fetch upstream
> git pull --rebase upstream master
> git log --oneline #check commits are gone 
> git push --force origin 10645
> ```

Thanks. I've removed those commits.

File: src/web/services/link.service.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ export class LinkService {
   JOIN_PAGE: string = '/join';
   STUDENT_HOME_PAGE: string = '/student/home';
   INSTRUCTOR_HOME_PAGE: string = '/instructor/home';
+  INSTRUCTOR_COURSE_DETAILS_PAGE: string = '/instructor/courses/details';
```
Comment (t-cheepeng) [inline src/web/services/link.service.ts:18]: typo? 

`COURSES` instead of `COURSE`

**Review by t-cheepeng — CHANGES_REQUESTED** [review]

File: src/web/services/link.service.ts (hunk 1 above)
Comment (g-lilian) [reply src/web/services/link.service.ts]: Noted and fixed!

**Review by t-cheepeng — APPROVED** [review]

**Review by ccyccyccy — COMMENTED** [review]
Is the link supposed to be a masquerade link? SInce the link currently doesn't work as an admin.
image

**Comment by g-lilian** [comment]
> Is the link supposed to be a masquerade link? SInce the link currently doesn't work as an admin.
> image
Oh.. How does the masquerade work? It linked correctly when I signed in as admin using the dev server.

**Comment by ccyccyccy** [comment]
When logged in as administrator, masquerade mode can also be used to impersonate instructors and students by adding `user=username` to the URL e.g `http://localhost:8080/web/student/home?user=johnKent.`

> Link student name to the course of that student in the course instructor (i.e., masquerade as the instructor of the course) 

The issue mentions that you need to append the query param.

Below is what I am seeing. Can you provide a gif if your solution works currently?

Screen Recording 2020-12-19 at 3 58 15 PM

**Comment by g-lilian** [comment]
Yup here is my gif. As you can see in the browser address bar, the course details page is generated and I can view it as an admin.
adminsearch

**Comment by ccyccyccy** [comment]
Might be something wrong at my end then. @t-cheepeng Can you confirm that it works without needing masquerade mode?

**Comment by wkurniawan07** [comment]
@ccyccyccy there's nothing wrong on your side. The link definitely needs masquerade mode. In fact, it is the main difficulty of the issue: finding out which instructor to be masqueraded as.

@g-lilian the logged in user you're using must be the instructor of the course being visited (`test-course-id`) and at the same time being admin, that's why you could get it to work without masquerade mode.

@damithc how would linking to the course details page help? I suspect you're thinking of some other page but it came across as that.

**Comment by g-lilian** [comment]
@wkurniawan07 
Oh, I see. So that's why it works for me. Does masquerade mode just involve appending `user=username` as a param? i.e. the generated link should instead be `.../courses/details?courseid=test-course-id&user=test-user` where test-user must be the instructor of the course being visited?

**Comment by damithc** [comment]
> @damithc how would linking to the course details page help? I suspect you're thinking of some other page but it came across as that.

Good point. The page I (as the admin) need to reach can vary based on the problem I'm trying to troubleshoot on behalf of a student user. For example, if a student says he can't see any team members, I need to check the 'student list' page to see if the team ID of this student vary slightly from other team members. The important thing is I should be able to switch easily to the masquerade mode for the instructor of the student. The exact page doesn't actually matter. Perhaps, linking to the student name to the 'student list' page is more intuitive?

**Comment by wkurniawan07** [comment]
> linking to the student name to the 'student list' page is more intuitive

That's even worse because student list page is essentially a combination of multiple tables in the course details page?

**Comment by damithc** [comment]
> > linking to the student name to the 'student list' page is more intuitive
> 
> That's even worse because student list page is essentially a combination of multiple tables in the course details page?

Worse in what sense?
How about 'student details' page of that student? That's specifically a page related to the student.

**Comment by t-cheepeng** [comment]
@ccyccyccy
Tested and the link does requires masquerade mode as mentioned

**Comment by damithc** [comment]
Any chance of salvaging this PR? Functionality-wise, I'm fine as long as it goes to an instructor page of the same course. My primary objective is to be able to masquerade an instructor of that course with one click. The current behavior of this PR seems good enough for that purpose.

**Comment by wkurniawan07** [comment]
> My primary objective is to be able to masquerade an instructor of that course with one click. The current behavior of this PR seems good enough for that purpose.

No. The current PR as is does not even introduce masquerading.

**Comment by damithc** [comment]
> No. The current PR as is does not even introduce masquerading.

Right. I misinterpreted @ccyccyccy 's comment.

Let's close this PR in that case so that someone else can take shot at it.

**Pull request closed without merging**